app/controllers/assessments_controller.rb

```diff
@@ -70,8 +70,20 @@ def new
     @threshold_hash = jurisdiction.hierarchical_symptomatic_condition.threshold_condition_hash
     @translations = @assessment.translations
     @contact_info = jurisdiction.contact_info
-    @lang = permitted_params[:lang] if %w[en es es-PR so fr].include?(params[:lang])
-    @lang = 'en' if @lang.nil? # Default to english
+
+    # For backwards-compatibility reasons, we still want to support the old 2-letter language codes
+    legacy_language_mapping = {
+      'en': 'eng',
+      'es': 'spa',
+      'es-PR': 'spa-PR',
+      'so': 'som',
+      'fr': 'fra'
+    }
+    unless params[:lang].nil?
```

Comment:
if we move the legacy mapping part to below line 85, and only try to match legacy codes if `@lang` is `nil` at that point, we can avoid doing an unnecessary lookup when `@lang` is one of the new codes

Reply:
good idea! changed the order.